src/Discord.Net.Core/Entities/Guilds/WelcomeScreen.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+
+namespace Discord;
+
+public class WelcomeScreen
+{
+    /// <summary>
+    ///     Gets the server description shown in the welcome screen. <see langword="null"/> if not set.
+    /// </summary>
+    public string Description { get; }
+
+    /// <summary>
+    ///     Gets the channels shown in the welcome screen, up to 5 channels.
+    /// </summary>
+    public IReadOnlyCollection<WelcomeScreenChannel> Channels { get; }
+
+    internal WelcomeScreen(string description, IReadOnlyCollection<WelcomeScreenChannel> channels)
+    {
+        Description = description;
+
+        Channels = channels;
+    }
```

Comment:
should `channels` be converted into an ImmutableArray to prevent downcasting here?

Reply:
good point. fixed.